### DISCUSSION TITLE
How to stop Rector from moving use statements above the top-level doc block?

### DISCUSSION THREAD

**Discussion by tchin25** [rectorphp/rector#6831] [Q&A]
Hello!

I'm currently trying to use Rector on a large project where every file has a license header in a doc block at the top of it like:
```php
<?php

/**
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License...
 */

use A;
use B;

/**
 * Some documentation about class D
 */
class D {}
```

But when I use Rector and it auto-imports a class `C`, it moves all the use statements above the license header:

```php
<?php

use A;
use B;
use C;

/**
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License...
 */
/**
 * Some documentation about class D
 */
class D {}
```

Is there a way to stop this from happening? Thanks!

**Comment by TomasVotruba** [comment]
Hi, thanks for reporting.

This is rather a bug. Could you add issue on the repository? The test into docblock class would be awesome :+1:

**Reply by tchin25** [reply]
Ok I made a new issue #6833
